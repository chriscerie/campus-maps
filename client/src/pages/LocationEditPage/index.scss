#location-edit-page-container {
  display: table;
  margin-top: 4em;
  margin-bottom: 10em;
}

#location-edit-page-inner {
  float: left;
  box-sizing: border-box;
  width: 100%;
  text-align: left;
}

#location-edit-page-inner p {
  margin-bottom: 2em;
}

#location-edit-page-inner input {
  display: block;
  margin: 0 0 1em;
  padding: 0.5em 0.8em;
  border: 1px solid #999;
  border-radius: 0.3em;
  width: 30em;
  max-width: 70%;
}

#location-edit-page-inner input:focus {
  outline: none;
  border-color: #2592ff;
  box-shadow: 0 0 3px rgba(52, 160, 255, 0.5);
}
