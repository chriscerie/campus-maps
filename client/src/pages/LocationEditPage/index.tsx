/* eslint-disable eqeqeq */
import { useSelector } from 'react-redux';
import { RootState } from '../../reducers';
import { useParams, Link } from 'react-router-dom';
import { useEffect } from 'react';
import { Container, Button, Grid } from '@mui/material';
import { Close } from '@mui/icons-material';
import { useForm, useFieldArray } from 'react-hook-form';
import axios from 'axios';
import { getLocationInfo } from '../../api/LocationAPI';
import { StatesList } from '../../types/StatesList';
import './index.scss';

function LocationEditPage() {
  const { id } = useParams<{ id: string }>();
  const mapInstance = useSelector((state: RootState) => state.mapInstance.map);
  const { register, control, setValue, handleSubmit, getValues } = useForm<{
    name: string;
    type: string;
    description: string;
    address1: string;
    address2: string;
    city: string;
    state: string;
    zip_code: string;
    rooms: Array<{
      room_name: string;
    }>;
  }>({
    shouldUnregister: false,
  });
  const { fields, append, remove } = useFieldArray({
    control,
    name: 'rooms',
  });

  // Set point
  useEffect(() => {
    if (mapInstance) {
      getLocationInfo(id, mapInstance).then((res) => {
        if (res) {
          setValue('name', res.name || '');
          setValue('type', res.type || '');
          setValue('description', res.description || '');
          setValue('address1', res.address1 || '');
          setValue('address2', res.address2 || '');
          setValue('city', res.city || '');
          setValue('state', res.state || '');
          setValue('zip_code', res.zip_code || '');
          setValue('rooms', res.rooms || [{ room_name: '' }]);
        }
      });
    }
  }, [id, mapInstance, setValue]);

  return (
    <Container maxWidth="md" id="location-edit-page-container">
      <div id="location-edit-page-inner">
        <h2>Update location details</h2>
        <p>All changes are subject to moderation.</p>
        <form>
          <ul>
            <li key="Name">
              <label>Name</label>
              <input
                disabled={true}
                {...register('name')}
                type="text"
                placeholder="Main Hall"
              />
            </li>
            <li key="Type">
              <label>Type</label>
              <input
                {...register('type')}
                type="text"
                placeholder="University Building"
              />
            </li>
            <li key="Description">
              <label>Description</label>
              <textarea {...register('description')} />
            </li>
            <li key="Address1">
              <label>Address 1</label>
              <input
                {...register('address1')}
                type="text"
                placeholder="123 Main St"
              />
            </li>
            <li key="Address2">
              <label>Address 2</label>
              <input
                {...register('address2')}
                type="text"
                placeholder="123 Main St"
              />
            </li>
            <li key="City">
              <label>City</label>
              <input
                {...register('city')}
                type="text"
                placeholder="San Francisco"
              />
            </li>

            <li key="State">
              <label>State</label>
              <select {...register('state')}>
                {Object.entries(StatesList).map(([key, state]) => {
                  return (
                    <option key={key} value={key}>
                      {state}
                    </option>
                  );
                })}
              </select>
            </li>

            <li key="ZIP">
              <label>ZIP Code</label>
              <input
                {...register('zip_code')}
                type="text"
                placeholder="11111"
              />
            </li>

            <li key="Rooms">
              <label>Room Numbers</label> <br />
              {fields.map((item, index) => (
                <Grid container item xs={12}>
                  <Grid item>
                    <input
                      {...register(`rooms.${index}.room_name` as const)}
                      type="text"
                      placeholder={'1111'}
                    />
                  </Grid>
                  {fields.length > 1 && index < fields.length - 1 && (
                    <Grid item>
                      <Button
                        id="remove-classroom-button"
                        onClick={() => {
                          if (fields.length > 1) {
                            remove(index);
                          }
                        }}
                      >
                        <Close id="close-icon" />
                      </Button>
                    </Grid>
                  )}
                </Grid>
              ))}
              <Button
                variant="contained"
                disableElevation
                id="location-edit-add-class-button"
                onClick={() => {
                  const lastRoom =
                    getValues('rooms')[getValues('rooms').length - 1];
                  if (lastRoom.room_name !== '') {
                    append({
                      room_name: '',
                    });
                  }
                }}
              >
                Add
              </Button>
            </li>
          </ul>
          <div style={{ marginTop: '2em' }}>
            <Button
              variant="contained"
              disableElevation
              id="location-edit-submit-button"
              onClick={handleSubmit((data) => {
                console.log(data);
                axios
                  .post(`/api/v1/locations/loc-edit/${id}`, data)
                  .then((res) => {
                    console.log(res);
                  })
                  .catch((err) => {
                    console.log(err);
                  });
              })}
            >
              Submit Changes
            </Button>
            <Link to={`/loc/${id}`} id="location-edit-cancel-link">
              Cancel
            </Link>
          </div>
        </form>
      </div>
    </Container>
  );
}

export default LocationEditPage;
