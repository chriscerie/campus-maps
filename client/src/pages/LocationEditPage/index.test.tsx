import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../reducers';
import LocationEditPage from './index';

describe('LocationEditPage', () => {
  it('should render successfully', () => {
    const renderer = ShallowRenderer.createRenderer();
    const tree = renderer.render(
      <Provider store={store}>
        <LocationEditPage />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
