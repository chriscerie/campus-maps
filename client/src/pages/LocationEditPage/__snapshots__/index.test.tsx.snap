// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationEditPage should render successfully 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-xdmu94-MuiContainer-root{width:100%;margin-left:auto;box-sizing:border-box;margin-right:auto;display:block;padding-left:16px;padding-right:16px;}@media (min-width:600px){.css-xdmu94-MuiContainer-root{padding-left:24px;padding-right:24px;}}@media (min-width:900px){.css-xdmu94-MuiContainer-root{max-width:900px;}}",
      }
    }
    data-emotion="css xdmu94-MuiContainer-root"
  />,
  <div
    className="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
    id="location-edit-page-container"
  >
    <div
      id="location-edit-page-inner"
    >
      <h2>
        Update location details
      </h2>
      <p>
        All changes are subject to moderation.
      </p>
      <form>
        <ul>
          <li>
            <label>
              Name
            </label>
            <input
              disabled={true}
              name="name"
              placeholder="Main Hall"
              type="text"
            />
          </li>
          <li>
            <label>
              Type
            </label>
            <input
              name="type"
              placeholder="University Building"
              type="text"
            />
          </li>
          <li>
            <label>
              Description
            </label>
            <textarea
              name="description"
            />
          </li>
          <li>
            <label>
              Address 1
            </label>
            <input
              name="address1"
              placeholder="123 Main St"
              type="text"
            />
          </li>
          <li>
            <label>
              Address 2
            </label>
            <input
              name="address2"
              placeholder="123 Main St"
              type="text"
            />
          </li>
          <li>
            <label>
              City
            </label>
            <input
              name="city"
              placeholder="San Francisco"
              type="text"
            />
          </li>
          <li>
            <label>
              State
            </label>
            <select
              name="state"
            >
              <option
                value=""
              >
                Select One
              </option>
              <option
                value="AL"
              >
                Alabama
              </option>
              <option
                value="AK"
              >
                Alaska
              </option>
              <option
                value="AZ"
              >
                Arizona
              </option>
              <option
                value="AR"
              >
                Arkansas
              </option>
              <option
                value="CA"
              >
                California
              </option>
              <option
                value="CO"
              >
                Colorado
              </option>
              <option
                value="CT"
              >
                Connecticut
              </option>
              <option
                value="DE"
              >
                Delaware
              </option>
              <option
                value="DC"
              >
                District of Columbia
              </option>
              <option
                value="FL"
              >
                Florida
              </option>
              <option
                value="GA"
              >
                Georgia
              </option>
              <option
                value="HI"
              >
                Hawaii
              </option>
              <option
                value="ID"
              >
                Idaho
              </option>
              <option
                value="IL"
              >
                Illinois
              </option>
              <option
                value="IN"
              >
                Indiana
              </option>
              <option
                value="IA"
              >
                Iowa
              </option>
              <option
                value="KS"
              >
                Kansas
              </option>
              <option
                value="KY"
              >
                Kentucky
              </option>
              <option
                value="LA"
              >
                Louisiana
              </option>
              <option
                value="ME"
              >
                Maine
              </option>
              <option
                value="MD"
              >
                Maryland
              </option>
              <option
                value="MA"
              >
                Massachusetts
              </option>
              <option
                value="MI"
              >
                Michigan
              </option>
              <option
                value="MN"
              >
                Minnesota
              </option>
              <option
                value="MS"
              >
                Mississippi
              </option>
              <option
                value="MO"
              >
                Missouri
              </option>
              <option
                value="MT"
              >
                Montana
              </option>
              <option
                value="NE"
              >
                Nebraska
              </option>
              <option
                value="NV"
              >
                Nevada
              </option>
              <option
                value="NH"
              >
                New Hampshire
              </option>
              <option
                value="NJ"
              >
                New Jersey
              </option>
              <option
                value="NM"
              >
                New Mexico
              </option>
              <option
                value="NY"
              >
                New York
              </option>
              <option
                value="NC"
              >
                North Carolina
              </option>
              <option
                value="ND"
              >
                North Dakota
              </option>
              <option
                value="OH"
              >
                Ohio
              </option>
              <option
                value="OK"
              >
                Oklahoma
              </option>
              <option
                value="OR"
              >
                Oregon
              </option>
              <option
                value="PA"
              >
                Pennsylvania
              </option>
              <option
                value="RI"
              >
                Rhode Island
              </option>
              <option
                value="SC"
              >
                South Carolina
              </option>
              <option
                value="SD"
              >
                South Dakota
              </option>
              <option
                value="TN"
              >
                Tennessee
              </option>
              <option
                value="TX"
              >
                Texas
              </option>
              <option
                value="UT"
              >
                Utah
              </option>
              <option
                value="VT"
              >
                Vermont
              </option>
              <option
                value="VA"
              >
                Virginia
              </option>
              <option
                value="WA"
              >
                Washington
              </option>
              <option
                value="WV"
              >
                West Virginia
              </option>
              <option
                value="WI"
              >
                Wisconsin
              </option>
              <option
                value="WY"
              >
                Wyoming
              </option>
            </select>
          </li>
          <li>
            <label>
              ZIP Code
            </label>
            <input
              name="zip_code"
              placeholder="11111"
              type="text"
            />
          </li>
          <li>
            <label>
              Room Numbers
            </label>
             
            <br />
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-1diear7-MuiButton-root{font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 16px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#fff;background-color:#1976d2;box-shadow:0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);box-shadow:none;}.css-1diear7-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:#1565c0;box-shadow:0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);}@media (hover: none){.css-1diear7-MuiButton-root:hover{background-color:#1976d2;}}.css-1diear7-MuiButton-root:active{box-shadow:0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);}.css-1diear7-MuiButton-root.Mui-focusVisible{box-shadow:0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);}.css-1diear7-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);box-shadow:none;background-color:rgba(0, 0, 0, 0.12);}.css-1diear7-MuiButton-root:hover{box-shadow:none;}.css-1diear7-MuiButton-root.Mui-focusVisible{box-shadow:none;}.css-1diear7-MuiButton-root:active{box-shadow:none;}.css-1diear7-MuiButton-root.Mui-disabled{box-shadow:none;}",
                }
              }
              data-emotion="css 1diear7-MuiButton-root"
            />
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-zddlty-MuiButtonBase-root-MuiButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 16px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#fff;background-color:#1976d2;box-shadow:0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root::-moz-focus-inner{border-style:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-zddlty-MuiButtonBase-root-MuiButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-zddlty-MuiButtonBase-root-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:#1565c0;box-shadow:0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);}@media (hover: none){.css-zddlty-MuiButtonBase-root-MuiButton-root:hover{background-color:#1976d2;}}.css-zddlty-MuiButtonBase-root-MuiButton-root:active{box-shadow:0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-focusVisible{box-shadow:0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);box-shadow:none;background-color:rgba(0, 0, 0, 0.12);}.css-zddlty-MuiButtonBase-root-MuiButton-root:hover{box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-focusVisible{box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root:active{box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-disabled{box-shadow:none;}",
                }
              }
              data-emotion="css zddlty-MuiButtonBase-root-MuiButton-root"
            />
            <button
              className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-zddlty-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              id="location-edit-add-class-button"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Add
            </button>
          </li>
        </ul>
        <div
          style={
            Object {
              "marginTop": "2em",
            }
          }
        >
          <button
            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-zddlty-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            id="location-edit-submit-button"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Submit Changes
          </button>
          <a
            href="/loc/1111"
            id="location-edit-cancel-link"
            onClick={[Function]}
          >
            Cancel
          </a>
        </div>
      </form>
    </div>
  </div>,
]
`;
